File: frontend/src/types/weekly-plan.ts
/** Status of a weekly plan */
export type PlanStatus = "completed" | "in-progress" | "upcoming";

/** A single task within a weekly plan */
export interface WeeklyTask {
  id: string;
  description: string;
  estimated_hours: number;
  skill_name: string;
  completion_criterion: string;
  completed: boolean;
}

/** A weekly plan containing tasks */
export interface WeeklyPlan {
  id: string;
  roadmap_id: string;
  week_number: number;
  status: PlanStatus;
  tasks: WeeklyTask[];
  is_practical_milestone: boolean;
}

/** Result from the delay adjustment endpoint */
export interface AdjustDelayResult {
  plans: WeeklyPlan[];
  message: string;
}

File: backend/app/schemas/weekly_plan.py
"""Pydantic models for weekly learning plans."""

from uuid import UUID

from pydantic import BaseModel, Field

from .common import PlanStatus


class WeeklyTask(BaseModel):
    id: UUID
    description: str
    estimated_hours: float
    skill_name: str
    completion_criterion: str
    completed: bool


class WeeklyPlan(BaseModel):
    id: UUID
    roadmap_id: UUID
    week_number: int
    status: PlanStatus
    tasks: list[WeeklyTask] = Field(min_length=3, max_length=7)
    is_practical_milestone: bool


class RoadmapCompletionSummary(BaseModel):
    """Response schema returned when the final plan is completed."""

    roadmap_id: UUID
    is_complete: bool
    total_weeks: int
    skills_acquired: list[str]
    message: str
